lib/eslint/eslint.js

```diff
@@ -314,7 +357,37 @@ async function loadFlatConfigFile(filePath) {
         delete require.cache[filePath];
     }
 
-    const config = (await import(fileURL)).default;
+    const isTS = isFileTS(filePath) && hasUnstableTSConfigFlag;
+
+    const isBun = isRunningInBun();
+
+    const isDeno = isRunningInDeno();
+
+    if (isTS && !isDeno && !isBun) {
+
+        const createJiti = await import("jiti").then(jitiModule => jitiModule.default, () => {
+            throw new Error("The 'jiti' library is required for loading TypeScript configuration files. Make sure to install it.");
+        });
+
+        /*
+         * Disabling `moduleCache` allows us to reload a
+         * config file when the last modified timestamp changes.
+         */
+
+        const jiti = createJiti(__filename, { moduleCache: false });
+
+        if (typeof jiti?.import !== "function") {
+            throw new TypeError("You are using an outdated version of the 'jiti' library. Please update to the latest version of 'jiti' to ensure compatibility and access to the latest features.");
```

Comment:
Can we make this a simple `Error`? It's not apparent to users that we are checking the type of `jiti?.import`, and actually it seems an implementation detail.

```suggestion
            throw new Error("You are using an outdated version of the 'jiti' library. Please update to the latest version of 'jiti' to ensure compatibility and access to the latest features.");
```